src/graph_export/graph_export.hpp

```diff
@@ -0,0 +1,34 @@
+//****************************************************************************
+// Copyright 2025 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+#include <string>
+
+namespace ovms {
+struct PluginConfigSettingsImpl;
+struct GraphSettingsImpl;
+class Status;
+
+class GraphExport {
+protected:
+    std::string graphString;
```

Comment:
Who will create graphExporter and where it will be used?
Holding GraphSettings in memory will be lighter & we could delay creating graph file until actually createGraphFile is used. Ideally we would not store either of those & just produce graphString on the fly when needed.